oauth2/src/main/java/com/jagrosh/jdautilities/oauth2/entities/impl/OAuth2ClientImpl.java

```diff
@@ -0,0 +1,245 @@
+                Scope[] scopes = new Scope[scopeStrings.length];
+                for(int i = 0; i < scopeStrings.length; i++)
+                {
+                    scopes[i] = Scope.from(scopeStrings[i]);
+                }
+
+                return sessionController.createSession(new SessionData(identifier,
+                    body.getString("access_token"), body.getString("refresh_token"),
+                    body.getString("token_type"), OffsetDateTime.now().plusSeconds(body.getInt("expires_in")), scopes));
+            }
+        };
+    }
+
+    @Override
+    public OAuth2Action<OAuth2User> getUser(Session session)
+    {
+        Checks.notNull(session, "Session");
+        return new OAuth2Action<OAuth2User>(this, Method.GET, OAuth2URL.CURRENT_USER.compile()) {
+
+            @Override
+            protected Headers getHeaders()
+            {
+                return Headers.of("Authorization", generateAuthorizationHeader(session));
+            }
+
+            @Override
+            protected OAuth2User handle(Response response) throws IOException
+            {
+                if(!response.isSuccessful())
+                    throw failure(response);
+                JSONObject body = new JSONObject(new JSONTokener(Requester.getBody(response)));
+                return new OAuth2UserImpl(OAuth2ClientImpl.this, session, body.getLong("id"),
+                    body.getString("username"), body.getString("discriminator"),
+                    body.optString("avatar", null), body.optString("email", null),
+                    body.optBoolean("verified", false), body.getBoolean("mfa_enabled"));
+            }
+        };
+    }
+
+    @Override
+    public OAuth2Action<List<OAuth2Guild>> getGuilds(Session session)
+    {
+        Checks.notNull(session, "session");
+        if(!Scope.contains(session.getScopes(), Scope.GUILDS))
+            throw new MissingScopeException("get guilds for a Session", Scope.GUILDS);
+        return new OAuth2Action<List<OAuth2Guild>>(this, Method.GET, OAuth2URL.CURRENT_USER_GUILDS.compile()) {
+            @Override
+            protected Headers getHeaders()
+            {
+                return Headers.of("Authorization", generateAuthorizationHeader(session));
+            }
+
+            @Override
+            protected List<OAuth2Guild> handle(Response response) throws IOException
+            {
+                if(!response.isSuccessful())
+                    throw failure(response);
+
+                JSONArray body = new JSONArray(new JSONTokener(Requester.getBody(response)));
+                List<OAuth2Guild> list = new LinkedList<>();
+                JSONObject obj;
+                for(int i = 0; i < body.length(); i++)
+                {
+                    obj = body.getJSONObject(i);
+                    list.add(new OAuth2GuildImpl(OAuth2ClientImpl.this, obj.getLong("id"),
+                        obj.getString("name"), obj.getString("icon"), obj.getBoolean("owner"),
```

Comment:
Isn't it better to use `obj.optString("icon", nul)` because the server icon can be null and it will throw this error: 
```
org.json.JSONException: JSONObject["icon"] not a string.
	at org.json.JSONObject.getString(JSONObject.java:810)
	at com.jagrosh.jdautilities.oauth2.entities.impl.OAuth2ClientImpl$3.handle(OAuth2ClientImpl.java:185)
	at com.jagrosh.jdautilities.oauth2.entities.impl.OAuth2ClientImpl$3.handle(OAuth2ClientImpl.java:165)
	at com.jagrosh.jdautilities.oauth2.requests.OAuth2Requester.submitSync(OAuth2Requester.java:95)
	at com.jagrosh.jdautilities.oauth2.requests.OAuth2Action.complete(OAuth2Action.java:147)
```

Reply:
Fixed with merge of #67 

Thanks for the help @duncte123! 😃